action-server/actions/daily_ci_feel_no_change_form.py

```diff
@@ -0,0 +1,153 @@
+from rasa_sdk.executor import CollectingDispatcher
+from rasa_sdk.forms import FormAction
+
+from actions.form_helper import request_next_slot
+
+FORM_NAME = "daily_ci_feel_no_change_form"
+
+LAST_SYMPTOMS_SLOT = "last_symptoms"
+
+SYMPTOMS_SLOT = "symptoms"
+HAS_FEVER_SLOT = "has_fever"
+HAS_DIFF_BREATHING_SLOT = "has_diff_breathing"
+HAS_COUGH_SLOT = "has_cough"
+
+
+class DailyCiFeelNoChangeForm(FormAction):
+    def name(self) -> Text:
+
+        return FORM_NAME
+
+    @staticmethod
+    def required_slots(tracker: Tracker) -> List[Text]:
+        """A list of required slots that the form has to fill"""
+
+        slots = [HAS_FEVER_SLOT, HAS_COUGH_SLOT]
+
+        last_symptoms = tracker.get_slot(LAST_SYMPTOMS_SLOT)
+
+        if last_symptoms == "moderate":
+            slots.append(HAS_DIFF_BREATHING_SLOT)
+
+        return slots
+
+    def slot_mappings(self) -> Dict[Text, Union[Dict, List[Dict]]]:
+        return {
+            HAS_FEVER_SLOT: [
+                self.from_intent(intent="affirm", value=True),
+                self.from_intent(intent="deny", value=False),
+            ],
+            HAS_COUGH_SLOT: [
+                self.from_intent(intent="affirm", value=True),
+                self.from_intent(intent="deny", value=False),
+            ],
+            HAS_DIFF_BREATHING_SLOT: [
+                self.from_intent(intent="affirm", value=True),
+                self.from_intent(intent="deny", value=False),
+            ],
+        }
+
+    def request_next_slot(
+        self,
+        dispatcher: CollectingDispatcher,
+        tracker: Tracker,
+        domain: Dict[Text, Any],
+    ) -> Optional[List[EventType]]:
+        return request_next_slot(
+            self, dispatcher, tracker, domain, self._utter_ask_slot_template
+        )
+
+    def _utter_ask_slot_template(self, slot: str, tracker: Tracker) -> Optional[str]:
+        if slot in [
+            HAS_FEVER_SLOT,
+            HAS_DIFF_BREATHING_SLOT,
+            HAS_COUGH_SLOT,
+        ]:
+            return f"utter_ask_daily_ci__feel_no_change__{slot}"
+
+        return None
+
+    def validate_has_fever(
+        self,
+        value: Text,
+        dispatcher: CollectingDispatcher,
+        tracker: Tracker,
+        domain: Dict[Text, Any],
+    ) -> Dict[Text, Any]:
+        if value == True:
+            dispatcher.utter_message(template="utter_daily_ci__acknowledge_fever")
+            dispatcher.utter_message(template="utter_daily_ci__take_acetaminophen")
+            dispatcher.utter_message(template="utter_daily_ci__avoid_ibuprofen")
+        else:
+            dispatcher.utter_message(
+                template="utter_daily_ci__feel_no_change__acknowledge_no_fever"
+            )
+
+        return {HAS_FEVER_SLOT: value}
+
+    def validate_has_cough(
+        self,
+        value: Text,
+        dispatcher: CollectingDispatcher,
+        tracker: Tracker,
+        domain: Dict[Text, Any],
+    ) -> Dict[Text, Any]:
+        if value == True:
```

Comment:
We should use `is True` instead of `==`. It won't change anything but the stricter is always cleaner.